src/components/Layout/NestedScrollLayoutContext.tsx

```diff
@@ -0,0 +1,38 @@
+import { createContext, PropsWithChildren, useEffect, useMemo, useRef } from "react";
+import { Css, Only, Xss } from "src/Css";
+
+interface NestedScrollLayoutContextProps {
+  scrollableEl: HTMLElement;
+}
+
+export const NestedScrollLayoutContext = createContext<NestedScrollLayoutContextProps>({
+  scrollableEl: document.createElement("div"),
+});
+
+// type NestedScrollLayoutXss = Xss<Margin | Padding | "flexGrow" | "flexShrink">;
+type NestedScrollLayoutXss = Xss<any>;
```

Comment:
I decided that any styles should be able to apply to the wrapping element here. For example, the TestSideNav component needs to be able to have some of its content scrollable, while the top part fixed. Though, the whole SideNav component can be toggled in/out of view, so we need to wrap that transition logic around everything within `NestedScrollLayoutContextProvider`'s root element.